src/components/Register/Step1.tsx

```diff
@@ -0,0 +1,133 @@
+import styled from "@emotion/styled";
+import { IcBack, IcCheckGrey, IcCheckWhiteGrey, IcNext, IcCheckMintWhite, IcCheckMint } from "../../assets";
+import Button from "../Common/Button/Button";
+import { Register } from "../../constant";
+import { useState } from "react";
+import CheckBox from "./CheckBox";
+import StepIcon from "../../utils/StepIcon";
+import { useNavigate } from "react-router-dom";
+
+interface AgreeTextProps {
+  isNecessary: boolean;
+}
+
+function Step1() {
+  const navigate = useNavigate();
+  const [checkedItems, setCheckedItems] = useState<string[]>([]);
+
+  const data = [
+    { text: "[필수] TinU 서비스 이용약관", key: "necessary1" },
+    { text: "[필수] 개인정보 수집 및 이용 동의", key: "necessary2" },
+    { text: "[선택] 광고성 정보 수신 동의", key: "optional" },
+  ];
+
+  const handleNextStep = () => {
+    navigate("/register/step2");
+  };
+
+  // 전체 선택
+  const handleAllChecked = (checked: boolean) => {
+    setCheckedItems(checked ? data.map((item) => item.key) : []);
+  };
+
+  // 개별 선택
+  const handleSingleChecked = (checked: boolean, key: string) => {
+    setCheckedItems((prev) => (checked ? [...prev, key] : prev.filter((item) => item !== key)));
+  };
+
+  return (
+    <Step1Container>
+      <Step1Wrapper>
+        <IcBack />
+        <StepIcon step={1} />
+        <TitleWrapper dangerouslySetInnerHTML={{ __html: Register.TEXT_STEP1 }} />
```

Comment:
`dangerouslySetInnerHTML`은 의도된 부분일까요?? 요거 사용하면 XSS 공격에 취약하다는 위험성이 있어요!!
`DOMPurify`와 함께 사용하면 안전하게 사용할 수 있어요.

```ts
  const sanitizedData = () => ({
    __html: DOMPurify.sanitize(Register.TEXT_STEP1)
  })
  ...
  
 <TitleWrapper dangerouslySetInnerHTML={sanitizedData()} />
```

이렇게 하면 잠재적인 악성코드를 추적해서 안전하게 데이터를 출력하게 돼요.
그래도 `dangerouslySetInnerHTML`를 사용할 명확한 이유가 없다면 잠재적으로 보안상 허점이 존재할 수 있기 때문에 지워주는 게 좋다고 생각해요!!

```suggestion
        <TitleWrapper>{Register.TEXT_STEP1}<TitleWrapper />
```